component/prometheus/operator/common/crdmanager.go

```diff
@@ -266,6 +275,20 @@ func (c *crdManager) runInformers(restConfig *rest.Config, ctx context.Context)
 		if ls != labels.Nothing() {
 			opts.DefaultLabelSelector = ls
 		}
+
+		// TODO: Remove custom opts.Mapper when sigs.k8s.io/controller-runtime >= 0.17.0 as `NewDynamicRESTMapper` is the default in that version
+		var err error
+		opts.HTTPClient, err = rest.HTTPClientFor(restConfig)
```

Comment:
I think this assignment is a bit premature since if there are errors, `opts` attribute still keep the erroneous client

Reply:
The code returns on error just below so `opts` will be discarded in that scenario regardless.

Reply:
Ah, good catch. LGTM then